File: lib/domain/agricultural_task/entities/crop_task_template_link_entity.rb
# frozen_string_literal: true

module Domain
  module AgriculturalTask
    module Entities
      class CropTaskTemplateLinkEntity
        attr_reader :id, :agricultural_task_id, :crop_id

        def initialize(id:, agricultural_task_id:, crop_id:)
          @id = id
          @agricultural_task_id = agricultural_task_id
          @crop_id = crop_id
        end
      end
    end
  end
end

File: crates/agrr-domain/src/agricultural_task/entities/crop_task_template_link_entity.rs
/// Ruby: `Domain::AgriculturalTask::Entities::CropTaskTemplateLinkEntity`
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct CropTaskTemplateLinkEntity {
    pub crop_id: i64,
}

impl CropTaskTemplateLinkEntity {
    pub fn new(crop_id: i64) -> Self {
        Self { crop_id }
    }
}
